datafusion/optimizer/src/optimizer.rs

```diff
@@ -384,9 +396,26 @@ impl Optimizer {
                     // rule handles recursion itself
                     None => optimize_plan_node(new_plan, rule.as_ref(), config),
                 }
-                // verify the rule didn't change the schema
                 .and_then(|tnr| {
-                    assert_schema_is_the_same(rule.name(), &starting_schema, &tnr.data)?;
+                    // run checks optimizer invariant checks, per pass
```

Comment:
what did you mean exactly here?

Reply:
The correct statement is `// run checks optimizer invariant checks, per optimizer rule applied`. And then multiple rules are applied per pass. Statement is fixed. Thanks!.